test/org/sagebionetworks/bridge/services/SchedulePlanServiceMockTest.java

```diff
@@ -0,0 +1,123 @@
+package org.sagebionetworks.bridge.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.sagebionetworks.bridge.TestUtils;
+import org.sagebionetworks.bridge.dao.SchedulePlanDao;
+import org.sagebionetworks.bridge.dynamodb.DynamoSchedulePlan;
+import org.sagebionetworks.bridge.models.schedules.Activity;
+import org.sagebionetworks.bridge.models.schedules.Schedule;
+import org.sagebionetworks.bridge.models.schedules.SchedulePlan;
+import org.sagebionetworks.bridge.models.schedules.ScheduleType;
+import org.sagebionetworks.bridge.models.schedules.SimpleScheduleStrategy;
+import org.sagebionetworks.bridge.models.surveys.Survey;
+import org.sagebionetworks.bridge.validators.SchedulePlanValidator;
+
+public class SchedulePlanServiceMockTest {
```

Comment:
At least one of these tests (probably doNotUseIdentifierFromClient) should use multiple schedule groups (AB test strategy) with multiple activities, so we can verify that we process *all* surveys. Maybe throw in a non-survey just to make sure we handle that case.